### PR TITLE
bump okhttp to 3.12.1

### PR DESCRIPTION
**Pull request by dulmandakh**
Bump OkHTTP to 3.12.1.

Changelog:
----------
[Android] [Changed] - bump OkHTTP 3.12.1

Test Plan:
----------
CI is green

### CODE REVIEW

**Comment by dulmandakh** [comment]
@gengjiawen please help with proguard issue.

**Comment by gengjiawen** [comment]
I will take a look.

**Comment by gengjiawen** [comment]
@dulmandakh Try add those in https://github.com/facebook/react-native/blob/master/ReactAndroid/proguard-rules.pro
```proguard
-dontwarn javax.naming.**
-keep class javax.naming.** {*;}
-dontwarn org.conscrypt.**
-keep class org.conscrypt.** {*;}
-dontwarn sun.security.**
-keep class sun.security.** {*;}
```

File: ReactAndroid/build.gradle [hunk 1]
```diff
@@ -330,7 +330,9 @@ dependencies {
     api 'com.google.code.findbugs:jsr305:3.0.2'
     api "com.squareup.okhttp3:okhttp:${OKHTTP_VERSION}"
     api "com.squareup.okhttp3:okhttp-urlconnection:${OKHTTP_VERSION}"
-    api 'com.squareup.okio:okio:1.14.0'
+    api 'org.bouncycastle:bcprov-jdk15on:1.60'
```
Comment (gengjiawen) [inline ReactAndroid/build.gradle]: Why not just use `okhttp-tls`.

Comment (dulmandakh) [reply ReactAndroid/build.gradle]: I looked into https://github.com/square/okhttp/blob/master/CHANGELOG.md and added those lines

**Comment by hramos** [comment]
Just curious - what do we gain from bumping to 3.12.1?

**Comment by dulmandakh** [comment]
@hramos previous versions didn't have connection timeout, but timeouts for connect, read and write. Once landed I plan to create a PR to do that. But need to bump okhttp in FB repo too. Also this version added support for TLS 1.3, but requires more dependencies to work. I'll work on that too.

Thank you

**Pull request closed without merging**